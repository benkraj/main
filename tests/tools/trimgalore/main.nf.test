nextflow_pipeline {

    name "Test --tools trimgalore"
    script "main.nf"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
                test = true
                tools = 'trimgalore'
            }
        }

        then {
            assert workflow.success
            assert new File("$outputDir/trimgalore/test_pair/test_pair_1.fastq.gz_trimming_report.txt").exists()
            assert new File("$outputDir/trimgalore/test_pair/test_pair_1_val_1.fq.gz").exists()
            assert new File("$outputDir/trimgalore/test_pair/test_pair_2.fastq.gz_trimming_report.txt").exists()
            assert new File("$outputDir/trimgalore/test_pair/test_pair_2_val_2.fq.gz").exists()
            assert new File("$outputDir/trimgalore/test_single/test_single.fastq.gz_trimming_report.txt").exists()
            assert new File("$outputDir/trimgalore/test_single/test_single_trimmed.fq.gz").exists()
            assert new File("$outputDir/trimgalore/versions.yml").exists()
        }
    }
}
