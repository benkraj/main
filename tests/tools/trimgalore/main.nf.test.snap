{
    "Should run without failures": {
        "content": [
            "test_pair_1.fastq.gz_trimming_report.txt:md5,38e4cb17a6f76186f9258e73c41e780e",
            "test_pair_2.fastq.gz_trimming_report.txt:md5,dccc562204af99a6538a05142a4ab65a",
            "test_single.fastq.gz_trimming_report.txt:md5,0f125d0001c33e4bc698ca479fd4aa18",
            "versions.yml:md5,0b8a9dd8a66aada1ecb389eccbb5ed0e"
        ],
        "timestamp": "2023-02-09T11:38:05+0000"
    }
}